test/unit/controllers/EmailConfirmedControllerSpec.scala

```diff
@@ -67,14 +66,12 @@ class EmailConfirmedControllerSpec extends ControllerSpec with BeforeAndAfterEac
 
 
       "email found in cache, email is verified and update verified email is successful but saving timestamp fails" in withAuthorisedUser() {
-        when(mockEmailCacheService.fetchEmail(any())(any(), any())).thenReturn(Future.successful(Some(EmailStatus("abc@def.com"))))
-        when(mockEmailCacheService.fetchTimeStamp(any())(any(), any())).thenReturn(Future.successful(None))
-
+        when(mockEmailCacheService.fetch(any())(any(), any())).thenReturn(Future.successful(Some(CachedData("abc@def.com", None))))
         when(mockEmailVerificationService.isEmailVerified(meq("abc@def.com"))(any[HeaderCarrier])).thenReturn(Future.successful(Some(true)))
         when(mockUpdateVerifiedEmailService.updateVerifiedEmail(meq("abc@def.com"), meq("GB1234567890"))(any[HeaderCarrier]))
           .thenReturn(Future.successful(Some(true)))
         when(mockEmailCacheService.remove(meq(InternalId("internalId")))(any(), any())).thenReturn(Future.failed(new InternalServerException("")))
-        when(mockEmailCacheService.saveTimeStamp(meq(InternalId("internalId")), any[DateTime])(any(), any())).thenReturn(Future.failed(new InternalServerException("")))
+        when(mockEmailCacheService.save(meq(InternalId("internalId")), any[CachedData])(any(), any())).thenReturn(Future.failed(new InternalServerException("")))
```

Comment:
I leave this to you but it feels like we should at least check what value will "CachedData" hold

Reply:
ok